packages/hooks/useNumber/useNumber.ts

```diff
@@ -0,0 +1,62 @@
+import { removeThousandsDelimiter, thousandsDelimiter } from "@utils/number";
+import { useEffect, useState } from "react";
+
+type Props = {
+  initialValue?: string;
+  validators?: Array<(value: number) => void>;
+  delimiters?: boolean;
+};
+
+/**
+ * This is a custom hook that handles numbers in a text input with optional input validators.
+ *
+ * @param {Object} [config] Optional.
+ * @param {string} [config.initialValue] Initial value of the number text. Optional.
+ * @param {Function[]} [config.validators] Array of validator functions. A validator receives the number value as a number type. Optional.
+ * @param {boolean} [config.delimiters] Whether to use thousands delimiters. Optional.
+ */
+export default function useNumber(config?: Props) {
+  const { initialValue = "", validators, delimiters = true } = config || {};
+
+  const [value, setValue] = useState(initialValue);
+  const [error, setError] = useState("");
+
+  const onChange = (newVal: string) => {
+    if (!validators) {
+      setValue(
+        delimiters
+          ? thousandsDelimiter(removeThousandsDelimiter(newVal))
+          : newVal
+      );
+      return;
+    }
+
+    for (const validator of validators) {
+      try {
+        validator(Number(removeThousandsDelimiter(newVal)));
+        setError("");
+      } catch (error) {
+        setError((error as Error).message);
+        break;
+      }
+    }
+
+    setValue(
+      delimiters ? thousandsDelimiter(removeThousandsDelimiter(newVal)) : newVal
+    );
+  };
+
+  useEffect(() => {
+    onChange(initialValue);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
I don't think this useEffect is necessary. 
When the dependency array is empty, it runs once when mounted. Therefore, since initialValue is already assigned in `const [value, setValue] = useState(initialValue);` putting initialValue in useEffect feels redundant.

Reply:
This part is identical to that of the `useText` hook already in the package. I vaguely remember there was an issue with the initial value not applying for some reason. But I can't seem to replicate it now. I'll get rid of both instances of `useEffect` in `useText` and `useNumber`.